grpc-java/grpc-servlet-websocket-jakarta/src/main/java/io/grpc/servlet/web/websocket/WebsocketStreamImpl.java

```diff
@@ -253,6 +249,9 @@ public void writeTrailers(Metadata trailers, boolean headersSent, Status status)
             } catch (IOException e) {
                 throw Status.fromThrowable(e).asRuntimeException();
             }
+            transportState().runOnTransportThread(() -> {
```

Comment:
It is very likely that more calls need to be wrapped in this, I'm going to continue to work on this - this is the main reason I have this marked as draft.

Reply:
Confirmed that this is necessary, but no other writes need to be wrapped in the way I was thinking. There is another flow control detail that is not correct, but flow control doesn't yet do anything for this or grpc-servlet.